qiita_db/metadata_template/test/test_prep_template.py

```diff
@@ -1464,6 +1464,32 @@ def test_validate_template_warning_missing(self):
         exp.sort_index(axis=1, inplace=True)
         assert_frame_equal(obs, exp)
 
+    def test_delete_column(self):
+        pt = qdb.metadata_template.prep_template.PrepTemplate(1)
+        pt.delete_column('run_date')
+        self.assertNotIn('run_date', pt.categories())
+
+        # testing errors
+        with self.assertRaises(qdb.exceptions.QiitaDBColumnError):
+            pt.delete_column('barcode')
+        with self.assertRaises(qdb.exceptions.QiitaDBColumnError):
+            pt.delete_column('ph')
+
+    def test_delete_sample(self):
+        pt = qdb.metadata_template.prep_template.PrepTemplate(2)
+        pt.delete_sample('1.SKM5.640177')
+        self.assertNotIn('1.SKM5.640177', pt.keys())
+
+        pt = qdb.metadata_template.prep_template.PrepTemplate(1)
+        with self.assertRaises(qdb.exceptions.QiitaDBColumnError):
```

Comment:
This is the same test as before, since it first checks if the prep info file has been processed. By changing the error type as I suggested in my previous comment you will see that this test will fail (or at least it should).